src/readsums.c

```diff
@@ -56,18 +56,25 @@ static rs_result rs_loadsig_add_sum(rs_job_t *job, rs_strong_sum_t *strong)
     rs_signature_t      *sig = job->signature;
     rs_block_sig_t      *asignature;
 
-    sig->count++;
-    new_size = sig->count * sizeof(rs_block_sig_t);
+    if (sig->count <= job->stats.sig_blocks) {
+
+        sig->count++;
```

Comment:
Yep...because sig->count is set before in rs_loadsig_s_stronglen() to a total count when estimating the number of signatures. If signature size is not known before, this codepath increments it after adding every block of sums from the signature just like now.